pkgs/servers/home-assistant/default.nix

```
@@ -88,50 +81,80 @@ in with py.pkgs; buildPythonApplication rec {
+    # the auth.mfa_modules.totp module
+    aiohttp-cors ciso8601 defusedxml emoji hass-frontend pynacl pillow pyotp
+    pyqrcode sqlalchemy
   ] ++ componentBuildInputs ++ extraBuildInputs;
 
   # upstream only tests on Linux, so do we.
   doCheck = stdenv.isLinux;
 
   checkInputs = [
-    asynctest pytest pytest-aiohttp requests-mock hass-nabucasa netdisco pydispatcher
+    asynctest pytest pytest-aiohttp pytest_xdist requests-mock hass-nabucasa netdisco pydispatcher
   ];
 
   checkPhase = ''
     # the tests require the existance of a media dir
     mkdir /build/media
 
     # - components' dependencies are not included, so they cannot be tested
-    # - test_merge_id_schema requires pyqwikswitch
-    # - test_loader.py tries to load not-packaged dependencies
-    # - test_notify pyotp doesn't like the short mock keys
-    # - unclear why test_merge fails: assert merge_log_err.call_count != 0
+    # - test_loader.py: tries to load not-packaged dependencies
+    # - test_notify: pyotp doesn't like the short mock keys
+    # - test_merge_id_schema: requires pyqwikswitch
+    # - test_merge: unclear why test_merge fails: assert merge_log_err.call_count != 0
     # - test_setup_safe_mode_if_no_frontend: requires dependencies for components we have not packaged
-    py.test \
+    # - a bunch of tests fail when run in parallel:
+    #   - test_create_default_config
+    #   - test_ensure_config_exists_creates_config
+    #   - test_ensure_config_exists_uses_existing_config
+    #   - test_load_yaml_config_converts_empty_files_to_dict
+    #   - test_load_yaml_config_preserves_key_order
+    #   - test_load_yaml_config_raises_error_if_unsafe_yaml
+    #   - test_load_yaml_config_raises_error_if_not_dict
+    #   - test_load_yaml_config_raises_error_if_unsafe_yaml
+    pytest -n auto \
       --ignore tests/components \
       --ignore tests/test_loader.py \
       --ignore tests/auth/mfa_modules/test_notify.py \
-      -k "not test_setup_safe_mode_if_no_frontend and not test_merge_id_schema and not test_merge"
-
-    # Some basic components should be tested however
-    py.test \
-      tests/components/{api,config,configurator,demo,discovery,frontend,group,history} \
-      tests/components/{homeassistant,http,logger,script,shell_command,system_log,websocket_api}
+      -k "not test_setup_safe_mode_if_no_frontend \
+          and not test_merge_id_schema \
+          and not test_merge \
+          and not test_create_default_config \
+          and not test_ensure_config_exists_creates_config \
+          and not test_ensure_config_exists_uses_existing_config \
+          and not test_load_yaml_config_converts_empty_files_to_dict \
+          and not test_load_yaml_config_preserves_key_order \
+          and not test_load_yaml_config_raises_error_if_malformed_yaml \
+          and not test_load_yaml_config_raises_error_if_not_dict \
+          and not test_load_yaml_config_raises_error_if_unsafe_yaml"
```

using pytestCheckHook would make this look a less intimidating, not to mention you can use comments more effectively
```suggestion
  disabledTests = [
          "test_load_yaml_config_converts_empty_files_to_dict"
          "test_load_yaml_config_preserves_key_order"
          "test_load_yaml_config_raises_error_if_malformed_yaml"
          "test_load_yaml_config_raises_error_if_not_dict"
          "test_load_yaml_config_raises_error_if_unsafe_yaml"
        ];
```

---

I'm all for using the hook, I wonder why this was split into two pytest runs though.

The first one excludes `test/components/`, the second runs explicit components from with `test/components/`. Is that possible within a single pytest run?

---

Turns out: Yes, it is. WIP.

---

Done. Though, I fear I need to do some more test runs to iron out the list racy tests.